lib/Service/Appointments/BookingService.php

```diff
@@ -112,14 +120,31 @@ public function confirmBooking(Booking $booking, AppointmentConfig $config): Boo
 			throw new ClientException('Could not make sense of booking times');
 		}
 
-		$calendar = $this->calendarWriter->write($config, $startObj, $booking->getDisplayName(), $booking->getEmail(), $booking->getDescription());
+		// TODO: inject broker here and remove indirection
+		$this->eventDispatcher->dispatchTyped(
+			new AppointmentBookedEvent(
+				$booking,
+				$config,
+			)
+		);
+
+		$calendar = $this->calendarWriter->write(
+			$config,
+			$startObj,
+			$booking->getDisplayName(),
+			$booking->getEmail(),
+			$booking->getDescription(),
+			$config->getCreateTalkRoom() ? $booking->getTalkUrl() : $config->getLocation(),
```

Comment:
@miaulalala :eyes: an appointment with a Talk room is implicitly taking place online, so there is no location to enter for the organizer and we will write the Talk URL instead